Mathlib/RingTheory/Flat/Basic.lean

```diff
@@ -35,44 +37,48 @@ This result is not yet formalised.
 * `Module.Flat.directSum`: arbitrary direct sums of flat modules are flat
 * `Module.Flat.of_free`: free modules are flat
 * `Module.Flat.of_projective`: projective modules are flat
+* `Module.Flat.iff_rTensor_preserves_injectiveness`: a module is flat iff tensoring preserves
+  injectiveness.
 
 ## TODO
 
-* Show that tensoring with a flat module preserves injective morphisms.
-  Show that this is equivalent to be flat.
-  See <https://stacks.math.columbia.edu/tag/00HD>.
-  To do this, it is probably a good idea to think about a suitable
-  categorical induction principle that should be applied to the category of `R`-modules,
-  and that will take care of the administrative side of the proof.
 * Show that flatness is stable under base change (aka extension of scalars)
   For base change, it will be very useful to have a "characteristic predicate"
   instead of relying on the construction `A ⊗ B`.
   Indeed, such a predicate should allow us to treat both
   `A[X]` and `A ⊗ R[X]` as the base change of `R[X]` to `A`.
   (Similar examples exist with `Fin n → R`, `R × R`, `ℤ[i] ⊗ ℝ`, etc...)
 * Generalize flatness to noncommutative rings.
+
 -/
 
 
 universe u v w
 
 namespace Module
 
-open Function (Injective Surjective)
+open Function (Surjective)
 
 open LinearMap Submodule TensorProduct
 
-variable (R : Type u) (M : Type v) [CommRing R] [AddCommGroup M] [Module R M]
+ variable (R : Type u) (M : Type v) [CommRing R] [AddCommGroup M] [Module R M]
 
 /-- An `R`-module `M` is flat if for all finitely generated ideals `I` of `R`,
 the canonical map `I ⊗ M →ₗ M` is injective. -/
 class Flat : Prop where
-  out : ∀ ⦃I : Ideal R⦄ (_ : I.FG), Injective (TensorProduct.lift ((lsmul R M).comp I.subtype))
+  out : ∀ ⦃I : Ideal R⦄ (_ : I.FG),
+    Function.Injective (TensorProduct.lift ((lsmul R M).comp I.subtype))
 #align module.flat Module.Flat
 
+/-- An `R`-module is flat if for all injectives `R`-linear maps `L : N → N'`, `L ⊗ 𝟙 M` is also
+  injective. -/
+def Flat.rTensor_preserves_injectiveness : Prop :=
```

Comment:
What are doing here? For Lean you are defining a new notion `Flat.rTensor_preserves_injectiveness`.

Reply:
My thought is to give 
```
∀ ⦃N N' : Type v⦄ [AddCommGroup N] [AddCommGroup N'] [Module R N] [Module R N']
    (L : N →ₗ[R] N'), Function.Injective L → Function.Injective (L.rTensor M)
```
a name, so that I don't have to type such a long statement everytime. Then prove that `Flat.rTensor_preserves_injectiveness` iff `Flat`

Reply:
I was confused by the doc.

Anyway, do you think a separate definition is really useful? I mean, you are not providing any API for `rTensor_preserves_injectiveness` except that is equivalent to `Flat`: maybe you can show that `Flat` iff `∀ ⦃N N' : Type v⦄...` without giving a name to this notion. It seems to me that it just add a layer of abstraction.

Reply:
I have deleted this definition. It is not really useful, just save some typing